plugins/mysql/mysqlembed.cpp

```diff
@@ -427,6 +457,37 @@ class MySQLConnection : public CInterface
 CIArrayOf<MySQLConnection> MySQLConnection::globalCachedConnections;
 CriticalSection MySQLConnection::globalCacheCrit;
 
+Semaphore MySQLConnectionCloserThread::closing;
+
+int MySQLConnectionCloserThread::run()
+{
+    loop
+    {
+        if (closing.wait(10000))
+        {
+            break;
+        }
+        MySQLConnection::retireCache(60000);
```

Comment:
this and check period should probably be constants at the start of the file.  If we ever implement plugin options then they may want to become configurable.